src/assets/CollectiblesController.ts

```diff
@@ -236,14 +237,27 @@ export class CollectiblesController extends BaseController<
   ): Promise<CollectibleMetadata> {
     const tokenURI = this.getCollectibleApi(contractAddress, tokenId);
     let collectibleInformation: ApiCollectible;
-
     /* istanbul ignore if */
-    if (this.openSeaApiKey) {
-      collectibleInformation = await handleFetch(tokenURI, {
-        headers: { 'X-API-KEY': this.openSeaApiKey },
-      });
-    } else {
+    try {
       collectibleInformation = await handleFetch(tokenURI);
+    } catch (e) {
+      logOrRethrowError(e);
+
+      if (this.openSeaApiKey) {
+        collectibleInformation = await handleFetch(
+          `https://api.opensea.io/api/v1/asset/${contractAddress}/${tokenId}`,
```

Comment:
Nit: This could be moved into a `get___` function, to be more in-line with how we generate the `tokenURI`. Same with the `asset_contract` and `assets` APIs. Having a method for building each API call seems like a nice pattern.

Or it could be built into the existing function, with a flag to prefer OpenSea? :thinking: :shrug: